File: src/lib/scala/Slate.Cloud/src/main/scala/slate/cloud/aws/AwsSupport.scala
/**
<slate_header>
  url: www.slatekit.com
  git: www.github.com/code-helix/slatekit
  org: www.codehelix.co
  author: Kishore Reddy
  copyright: 2016 CodeHelix Solutions Inc.
  license: refer to website and/or github
  about: A Scala utility library, tool-kit and server backend.
  mantra: Simplicity above all else
</slate_header>
*/

package slate.cloud.aws

import com.amazonaws.auth.profile.ProfileCredentialsProvider
import com.amazonaws.auth.{BasicAWSCredentials, AWSCredentials}


trait AwsSupport {

  def credentials(key:String, secret:String): AWSCredentials =
  {
    new BasicAWSCredentials(key, secret)
  }


  def credentialsFromLogon(): AWSCredentials =
  {
    new ProfileCredentialsProvider().getCredentials
  }
}

File: src/lib/kotlin/slatekit-cloud/src/main/kotlin/slatekit/cloud/aws/AwsSupport.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.cloud.aws

import com.amazonaws.auth.AWSCredentials
import com.amazonaws.auth.BasicAWSCredentials
import com.amazonaws.auth.profile.ProfileCredentialsProvider

interface AwsSupport {

    fun credentials(key: String, secret: String): AWSCredentials = BasicAWSCredentials(key, secret)

    fun credentialsFromLogon(): AWSCredentials = ProfileCredentialsProvider().credentials
}
